### PR TITLE
Feat/add mmkv plugin

### PR DESCRIPTION
**Pull request by cyrilbo**
This PR adds a plugin for react-native-mmkv

- I reused what has been implemented by @jthoward64 for the AsyncStorage Plugin

## Test
Tested on
- [x] iOS simulator
- [x] android emulator

## UI
image

### Automatic update support
With https://github.com/mrousavy/react-native-mmkv/blob/master/docs/LISTENERS.md

### Multi storage support
MMKV supports multiple storages. We should be able to watch all of them without manually editing what is passed at `useMMKVDevTools`

### Code sharing with AsyncStorage and over KeyValue Storage solutions

webui has been duplicated from the AsyncStorage plugin. As as KeyValue storage solutions offer the same kind of features. I think we should create :
-  a generic webUI interface to harmonize the developer experience
- and shared communication layer between the device and the plugin webUI

image

### CODE REVIEW

**Review by Kudo — APPROVED** [review]
thanks for this awesome work. code looks good to me and just a nit change. after that we can merge and publish the plugin. you could also list you as an author in [readme](https://github.com/expo/dev-plugins/blob/main/README.md)

File: packages/react-native-mmkv/webui/src/usePluginStore.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import type { KeyValuePair } from '@react-native-async-storage/async-storage/lib/typescript/types';
+import { type EventSubscription } from 'expo/devtools';
+import { useCallback, useEffect, useState } from 'react';
+import { Method, MethodAck } from '../../methods';
+import { DevToolsPluginClient } from 'expo/build/devtools/DevToolsPluginClient';
```
[the remark this answers is not in the document]
Comment (cyrilbo) [reply packages/react-native-mmkv/webui/src/usePluginStore.ts]: DevToolsPluginClient is not exported at the moment. I made a PR https://github.com/expo/expo/pull/27125

[the remark this answers is not in the document]
Comment (cyrilbo) [reply packages/react-native-mmkv/webui/src/usePluginStore.ts]: Hi @Kudo ! I updated expo package and fixed the imports :)

**Comment by NomadicOx** [comment]
Just checking in here on possibility of merging. Really interested in giving this a go! Thanks for the efforts here!

File: packages/react-native-mmkv/webui/src/usePluginStore.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyrilbo) [reply packages/react-native-mmkv/webui/src/usePluginStore.ts]: @Kudo can we merge this PR ? 🙂

**Review by Kudo — APPROVED** [review]

**Comment by Kudo** [comment]
@cyrilbo sorry for long waiting. the plugin is published in `@dev-plugin/react-native-mmkv`. i was struggling to find bandwidth to support broader plugins and then realized that be honest it is to difficult for me to review all prs and publish fixes in a scale way. i think a better way is to host new plugins in individual's npm accounts. i'm turning this repository as a gallery for people to promote new plugins as described in the [README](https://github.com/expo/dev-plugins/blob/main/README.md#contributions).

at this moment, i'm happy to help the `@dev-plugins/react-native-mmkv` transition when you get a chance to create a plugin in your account. sorry for this inconvenience and thanks again for your awesome work.

**Comment by cyrilbo** [comment]
@cyrilbo thanks for your time 🙏 I'll look into this transition 🙂

**Pull request merged**